Add code for Contact-Based Routing (CBR)

SujataPal opened the pull request:
Hi,
I would like to contribute the code for CBR.

julianofischer commented:
Very nice, congratulations!
ASAP I will read this code.
I think @akeranen and @tk721 will consider adding it to the code after a review.

SujataPal commented:
Thank you Juliano.

On Mon, Mar 13, 2017 at 12:01 AM, Juliano Fischer Naves <
notifications@github.com> wrote:

> Very nice, congratulations!
> ASAP I will read this code.
> I think @akeranen <https://github.com/akeranen> and @tk721
> <https://github.com/tk721> will consider adding it to the code after a
> review.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/akeranen/the-one/pull/45#issuecomment-285963999>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIXjjEfgRW2I6ONflOVU_WhUhzKyIKYEks5rlDnvgaJpZM4MMHM9>
> .
>



-- 
Thanks & Regards
Sujata Pal, PhD
Assistant Professor, CSE
IIT Ropar, India

akeranen commented on src/routing/CBR.java:
```diff
@@ -0,0 +1,184 @@
+    @Override
+    public void update() {
+        super.update();
+        //if(SimClock.getIntTime() == 43200)
```
Good to clean up the code before merging. And there's actually the core.Debug.p method for debug prints.

akeranen commented on src/routing/CBR.java:
```diff
@@ -0,0 +1,184 @@
+    public void update() {
+        super.update();
+        //if(SimClock.getIntTime() == 43200)
+            //System.out.println("[" + this.getHost() + "]  list: " + this.nodesInf + "\n");
```
Same here

akeranen commented on src/routing/CBR.java:
```diff
@@ -0,0 +1,184 @@
+                    ArrayList<Integer> items = otherRouter.nodesInf.get(msgDst.getAddress());                   
+                   
+                    int encounter = items.get(0);
+                    if (encounter > maxEnc)
```
Style issue: opening curly brackets on the same line in ifs
(also below)

julianofischer commented:
Hi @SujataPal, maybe you would like to comment your source code. Would be nice!

julianofischer commented:
I think renaming CBR.java to CBRRouter.java is a good idea since all routing classes are named following this rule.

SujataPal commented:
Hi Ari/Juliano,

I have changed the code according to your suggestion and renamed CBR.java to CBRRouter.java as per Juliano. 

Thank you,
Sujata
